modules/unsupported/jdbc-ng/jdbc-hana/src/main/java/org/geotools/data/hana/converter/BlobConverterFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.hana.converter;
+
+import java.sql.Blob;
+import org.geotools.factory.Hints;
+import org.geotools.util.Converter;
+import org.geotools.util.ConverterFactory;
+
+/**
+ * Converter factory for converting all classes implementing java.sql.Blob to byte[].
+ *
+ * @author Stefan Uhrig, SAP SE
+ */
+public class BlobConverterFactory implements ConverterFactory {
+
+    @Override
+    public Converter createConverter(Class<?> source, Class<?> target, Hints hints) {
+        if (byte[].class.equals(target) && Blob.class.isAssignableFrom(source)) {
+            return new Converter() {
+                @Override
+                public <T> T convert(Object source, Class<T> target) throws Exception {
+                    if (source == null) {
+                        throw new IllegalArgumentException();
+                    }
+                    Blob blob = (Blob) source;
+                    long length = blob.length();
+                    if (length == 0) {
+                        return target.cast(new byte[0]);
+                    }
+                    if (length > Integer.MAX_VALUE) {
```

Comment:
If I understood this correctly, the goal here is too avoid the error:

```
Exception in thread "main" java.lang.OutOfMemoryError:
  Requested array size exceeds VM limit
```

... if the blob size is bigger than the maximum array possible index. Two comments here:

* the array maximum size varies from a VM to another (http://hg.openjdk.java.net/jdk7/jdk7/jdk/file/tip/src/share/classes/java/util/ArrayList.java#l190)

* not sure that simply returning ``NULL`` is the best option, I would at least log some debug message saying: ``Conversion not supported this by this converter, blob to big for this converter to handle.``.

Reply:
Actually I did that because the dimension expression of an array must be of type `int` and cannot be of type `long`. When casting from `long` to `int`, the upper 32 bits are just discarded. So this check is to verify that none of the upper 32 bits is set.

According to the documentation of `org.geotools.util.Converter`, both returning `null` and throwing an exception is fine to signal that the converter is not able to convert. Nevertheless, logging a debug message in case of a failure does not hurt.

I propose I address that in a separate pull request if you agree that a `Blob` to `byte[]` converter and a `Clob` to `String` converter in `org.geotools.util` makes sense.